plugins/module_utils/vmware.py

```diff
@@ -281,3 +287,44 @@ def get_datacenter_detailed(self, name):
 
         """
         return self.get_objs_by_name_or_moid([vim.Datacenter], name)
+
+    def get_datacenter_by_name(self, dc_name, fail_on_missing=False):
```

Comment:
@bardielle i added this method, but it similar to the one you added `get_datacenter_detailed`. I didnt see that method being used, so i was wondering if we should combine them?

Reply:
Yes I think we can combine them maybe we can filter out if returning detailed or only the moid